frontend/src/components/smart-contracts/hooks/useJob.ts

```diff
@@ -146,17 +147,68 @@ export const useFindJobs = () => {
         allJobs.push(jobData);
       }
 
-      setJobs(allJobs);
+      if (jobFilter) {
+        setJobs(filterJobs(allJobs, jobFilter));
+      } else {
+        setJobs(allJobs);
+      }
       setIsLoading(false);
     };
 
     if (account) {
       fetchJobs();
     }
-  }, [account, contracts]);
+  }, [account, contracts, jobFilter]);
 
   return {
     jobs,
     isLoading,
   };
 };
+
+const filterJobs = (jobs: IJobData[], jobFilter: IJobFilter) => {
+  const filteredJobs: IJobData[] = [];
+  const filterFields = jobFilter.fields;
+  for (const job of jobs) {
+    let anyFound = false;
+    Object.getOwnPropertyNames(filterFields).forEach((key) => {
+      if (anyFound) {
+        return;
+      }
+
+      switch (key) {
+        case 'id':
+          if (job.id === filterFields.id) {
+            anyFound = true;
+          }
+          break;
+        case 'supplier':
+          if (job.supplier === filterFields.supplier) {
+            anyFound = true;
+          }
+          break;
+        case 'engineer':
+          if (job.engineer === filterFields.engineer) {
+            anyFound = true;
+          }
+          break;
+        case 'state':
+          if (job.state === filterFields.state) {
+            anyFound = true;
+          }
+          break;
+        case 'paymentTokenName':
+          if (job.paymentTokenName === filterFields.paymentTokenName) {
+            anyFound = true;
+          }
+          break;
+      }
+    });
+
+    if (anyFound) {
+      filteredJobs.push(job);
+    }
+  }
```

Comment:
Fine to do it this way, but might want to consider `reduce` here? Also I think we could use 

```
const filteredJobs = jobs.reduce(
      (prev, curr) => {
        switch (key) {
          case 'id':
            if (job.id === filterFields.id) {
              prev.push(curr);
            }
            break;
          case 'supplier':
            if (job.supplier === filterFields.supplier) {
              prev.push(curr);
            }
            break;
          ... etc ...
        }
        return prev;
      },
      []
    );
```

Reply:
Good idea.  I couldn't get a reduce function to work cleanly, so I used an Array.filter function.